src/axom/quest/IntersectionShaper.hpp

```diff
@@ -1466,6 +1441,458 @@ class IntersectionShaper : public Shaper
+      /*level 12*/ 3.14159251669,  // diff=1.36898179903e-07
+      /*level 13*/ 3.14159261936,  // diff=3.4225411838e-08
+      /*level 14*/ 3.14159264503,  // diff=8.55645243547e-09
+    };
+    constexpr int MAX_LEVELS = sizeof(lut) / sizeof(double);
+    return (level < MAX_LEVELS) ? (lut[level] * radius * radius)
+                                : calcCircleArea(radius, level);
+  }
+
+  /*!
+   * \brief Iterate over line segments in the surface mesh and compute
+   *        truncated cone values with circles at given refinement and
+   *        add them all up. This will be the approximate revolved
+   *        volume that is possible with the current curve to line segment
+   *        refinement that was done.
+   *
+   * \param m The mint mesh that contains the line segments.
+   * \param level The circle refinement level.
+   *
+   * \note This function assumes that the line segments have been
+   *       processed and increase in x.
+   *
+   * \return The approximate revolved volume for the linearized curve
+   *         and circle refinement level.
+   */
+  double volume(const mint::Mesh* m, int level) const
+  {
+    const int numSurfaceVertices = m->getNumberOfNodes();
+    const int nSegments = numSurfaceVertices - 1;
+    const double* z = m->getCoordinateArray(mint::X_COORDINATE);
+    const double* r = m->getCoordinateArray(mint::Y_COORDINATE);
+    double vol_approx = 0.;
+    for(int seg = 0; seg < nSegments; seg++)
+    {
+      double r0 = r[seg];
+      double r1 = r[seg + 1];
+      double h = fabs(z[seg + 1] - z[seg]);
+      if(axom::utilities::isNearlyEqual(r0, r1, m_vertexWeldThreshold))
+      {
+        // cylinder
+        double A = circleArea(r0, level);
+
+        // Add the cylinder to the volume total.
+        vol_approx += A * h;
+      }
+      else
+      {
+        // truncated cone
+        double h2 = (r0 * h / (r0 - r1)) - h;
+
+        // Approximate cone volume
+        double A2 = circleArea(r0, level);
+        double A3 = circleArea(r1, level);
+        double approx_cone_vol = (1. / 3.) * (A2 * h + (A2 - A3) * h2);
+
+        // Add the truncated cone to the volume total.
+        vol_approx += approx_cone_vol;
+      }
+    }
+
+    return vol_approx;
+  }
+
+  /*!
+   * \brief Refine the shape to get under a certain errorPercent in the
+   *        linearized revolved volume. To do this, we may end up reloading
+   *        the shape.
+   *
+   * \param shape The shape to refine.
+   *
+   * \note The m_surfaceMesh will be set to a mesh that should be fine
+   *       enough that its linearized representation is close enough to
+   *       the specified error percent. m_level will be set to the circle
+   *       refinement level that let us meet the percent error target.
+   */
+  void refineShape(const klee::Shape& shape)
+  {
+    // If we are not refining dynamically, return.
+    if(m_percentError <= MINIMUM_PERCENT_ERROR ||
+       m_refinementType != RefinementDynamic)
+      return;
```

Comment:
```suggestion
    if(m_percentError <= MINIMUM_PERCENT_ERROR ||
       m_refinementType != RefinementDynamic)
    {
      return;
    }
```

Reply:
Added braces. Maybe "make style" should catch this.